ballerina/tests/test.bal

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/os;
+import ballerina/oauth2;
+import ballerina/test;
+import ballerina/time;
+
+configurable string clientId =  ?;
+configurable string clientSecret = ?;
+configurable string refreshToken = ?;
+
+configurable boolean useMockServer = os:getEnv("useMockServer") == "true";
```

Comment:
This should be the other way around. The default way should be to use the mock server. This will try to run live tests when there's no env defined with `useMockServer`, which is the case in GitHub actions.

```suggestion
configurable boolean useLiveServer = os:getEnv("USE_LIVE_SERVER") == "true";
```